jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -96,9 +96,13 @@ MlirToJlmConverter::ConvertBlock(::mlir::Block & block, rvsdg::Region & rvsdgReg
   }
 
   // The results of the region/block are encoded in the terminator operation
-  ::mlir::Operation * terminator = block.getTerminator();
```

Comment:
An MLIR block is the equivalent of a region in RVSDG in this case, as MLIR doesn't allow regions without at least one block. The Terminator is what encodes the results/outputs of a region, so not having a Terminator is a region without outputs, i.e., it should be dead. If we want to allow dead regions/structural-nodes to be generated in MLIR, then a Terminator with no results should be added to the block, e.g., ::mlir::rvsdg::ThetaResult or ::mlir::rvsdg::GammaResult.

I'm not 100% sure what the issue is with the new MLIR version. But if it is that there isn't always a Terminator in the block, then wouldn't it be better to modify the MLIR backend to assure that there is always a block being insterted in each an every block?